metatensor-core/include/metatensor.hpp

```diff
@@ -551,14 +562,14 @@ class Labels final: public NDArray<int32_t> {
 
     /// Labels can be move-assigned
     Labels& operator=(Labels&& other) noexcept {
-        NDArray<int32_t>::operator=(std::move(other));
-
-        this->names_ = std::move(other.names_);
-
         mts_labels_free(&labels_);
         this->labels_ = other.labels_;
+        assert(this->labels_.internal_ptr_ != nullptr);
```

Comment:
no big reason, just defensive programming. this will only fail if something with constructor goes very wrong. I'm adding it since I spent a lot of time debugging the constructors